### PR TITLE
[CDU] Add MCDU formatting helpers

### PR DESCRIPTION
**Pull request by beheh**
## Summary of Changes

In some cases, the MCDU requires multiple colors per cell. A good example is the INIT B page, where the user may edit the fuel, but not the number of minutes (which are computed). Hence the cell is split in blue and green, which is not supported out of the box by the MCDU template. There are other in-cell formatting changes

To fix this in the past pages, various pages use a little marker in their template (e.g. ` {smallFront}` and `{smallEnd}`) together with script that replaces these markers in the innerHTML with `<span>`s using the corresponding CSS classes. This is a bit ugly, especially as multiple pages need this.

As part of #1342, @tyler58546 built a little helper for the CDU code that generalized the markers and applies them everywhere by default. Unfortunately #1342 is pretty sizeable and it's not clear when this PR will land. Therefore, after a [conversation with @Lucky38i in Discord](https://discord.com/channels/738864299392630914/747622836381810739/775306984680587274) I'm pulling this into a standalone, separate PR that does nothing but introduce the formatting helper where it's possible.

This PR adjusts the pages that are already in the repository to use this new methods. I've tried searching the code base for all I could find. Please let me know if you find any others. Once this PR lands I'm happy to help out with any open PRs that want to rebase onto this (although it should be very straightforward).

All in all there are no expected user facing changes, so I do not think a CHANGELOG entry is necessary. This for developers and should simplify code in a few places and make it easier and faster to implement cells with special formatting in future.

### Notes for QA (once tagged with 'Ready to Test')

QA should verify that all MCDU pages are formatted as usual (and as expected), especially the ones touched as part of this PR:
- AIRWAYS (FPLN => LAT REV => AIRWAYS)
- IRS INIT (INIT => IRS INIT)
- INIT

## Screenshots (if necessary)

While nearly all changes are invisible, I cleaned up the IRS INIT page that was using a font that was manually defined and did not match the normal font sizes, especially after #1243. Replacing this caused the font sizes to be corrected again, and this is the only change that should be visible to players (left old, right new):

IRS INIT

## Additional context

PRs that roll their own templating or are forgoing dual formatting and would benefit from this:
- #1524
- #1597
- #1823
- #1845

PR #1342 (where this is from) would also have to be adjusted. These PRs could be rebased once this PR lands to use the new templates.

Pages we know that use split colors and would benefit from this:
- INIT B (various fuel entries)
- FUEL PRED (same)
- AIRWAYS TO (title)
- PERF page (O=/S=/F=)
- DUPLICATE NAMES (waypoint)
- VERT REV (title)
- LAT REV (title)
- HOLD AT (title)
- FPLN (split colors in right column)

Credit for the original work goes to @tyler58546. Also thanks to @Lucky38i for some help on this.

Discord username (if different from GitHub): BehEh#1234

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by beheh** [comment]
Keeping this a draft while I'm still testing pages and cleaning up.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js
Comment (Lucky38i) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js:49]: There's a tag here you forgot to add to the helper. You can set it to something like `IrsInit` in the helper. The span class used you can find in line 128

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js
Comment (Lucky38i) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js:131]: remove this once you add the new tag to the helper

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js
Comment (Lucky38i) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:130]: put this right before the `{end}` tag.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js
Comment (Lucky38i) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:144]: put this right before the `{end}` tag.

**Review by Lucky38i — CHANGES_REQUESTED** [review]
Please check all the pages in the CDU that use some form of formatting helper. These aren't the only ones

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js
Comment (beheh) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js:49]: This one seems so specific to this page, that I don't think it makes sense to move this to the general MCDU code.

**Comment by beheh** [comment]
Not as-is, but that's enough reason then to support the line headers as well, so I add it.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js
Comment (beheh) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IRSInit.js:49]: This has now been normalized to use `{small}` as well.

**Comment by beheh** [comment]
@tyler58546, if you could take a look at this please and confirm you're okay with the changes that would be great. I've added you as co-author to the commit.

**Comment by beheh** [comment]
Urgh, some linter messed up some files, hang on...

**Comment by Lucky38i** [comment]
In case anyone asks, this supports label, line and title elements. Provided the write tag and span class is added in the helper loop

**Review by Lucky38i — APPROVED** [review]

**Comment by beheh** [comment]
I'm not sure if @tyler58546 is at all available at the moment. At this point the code has been significantly refactored from Tyler's original code and the most significant is the choice of special token names. From my side this is all ready to go, and @Lucky38i who's initial suggestion this was has also approved this.

Do the maintainers want Tyler's approval for this, or can they reach them somehow? Landing this would unblock improvements and simplifications in a few open PRs: #1524, #1597 and #1823

We can also probably start QA'ing this to make sure nothing has broken.

**Review by Benjozork — APPROVED** [review]

**Comment by Benjozork** [comment]
QA, please just test if formatting of MCDU pages is correct and everything is functional.

**Comment by stevex77** [comment]
I will test. Btw. @Benjozork where I can obtain the QA template? I didn't find it here, so should I ask someone specific for it within QA discord group? Thanks.

**Comment by danbw9** [comment]
Name: danbw9#7028
Date of testing: 11/11/2020
Version of the sim: 1.10.11.0
PR Tested: #1830 
Object of test: Test out MCDU formatting helpers

Testing technique: Normal flight from C&D + focus on MCDU.

On the ground, went through normal MCDU setup, looking at all functions.

In CRZ, checking through all MCDU pages, and testing various normal functions. No unusual stuff noticed. 

Confirmed the slight change in font in IRS INIT page as listed above.

As per the notes for QA:
- Instead of importing F-PLAN, manually input DEP, DEST, SID, STAR and Route. Normal behaviour expected via AIRWAYS function.
- IRS INIT - normal indication on this page on the ground as well as in CRZ
- INIT A + INIT B fields and functions all fine. 

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: LGTM. No issues whatsoever. I did everything as normal and in addition I checked out the items mentioned specifically in the PR above. Hopefully this will be a good platform for other PRs to come.

**Comment by z0rgmeister** [comment]
Name: z0rgmeister#8972
Date of testing: 11/11/2020
Version of the sim: 1.10.11.0
PR Tested: #1830
Object of test: Test out MCDU formatting helpers as per PR

Testing technique: Full LEMD-LPPT flight from C&D focusing on MCDU usage.

Normal MCDU setup following DIFSRIP mnemonic method.

Verified pages that require special cell formatting as per PR, no issues noticed.

As per the notes for QA:

    Manually input FPLAN with SID,STARs and Airways. Observed regular behaviour.
    ALIGN pages OK.
    INIT B Fuel entries OK

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: Great addition, no issues and a new tidy look for all these pages.

**Comment by stevex77** [comment]
Name: SteveX77
Date of testing: 11/11/2020
Version of the sim: 1.10.11.0
PR Tested: #1830
Object of test: Test out MCDU formatting helpers

Testing technique:
Flight KMCO - KMIA (ILS27 via HENDN, no STAR), started C&D Aug 1 2020 at 22:00 UTC, taxi to 36R.
On GND crawled through and checked literaly all MCDU pages, including whole new MCDU menu.
Randomly checked some pages during various flight phases.
Checked IRS INIT page during IRS alignment and once aligned.
Checked both INIT A and B pages couple of time.

As per the notes for QA:
Especially focused on these:
AIRWAYS (FPLN => LAT REV => AIRWAYS)
IRS INIT (INIT => IRS INIT)
INIT

Issues (If Any): Found one, attached, but seems not to be connected to this PR (have the feeling that I've seen this before, event in vanilla version)

TEST01

Overall Rating: GOOD
Conclusions of the testing: LGTM, also noticed slight font changes in IRS INIT.

**Pull request merged**
